ts/store/reducers/wallet/transactions.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * Reducers, states, selectors and guards for the transactions
+ */
+import { none, Option, some } from "fp-ts/lib/Option";
+import { createSelector } from "reselect";
+import { WalletTransaction } from "../../../types/wallet";
+import {
+  SELECT_TRANSACTION_FOR_DETAILS,
+  TRANSACTIONS_FETCHED
+} from "../../actions/constants";
+import { Action } from "../../actions/types";
+import { GlobalState } from "../types";
+import { getSelectedCreditCardId } from "./cards";
+
+export type TransactionsState = Readonly<{
+  transactions: ReadonlyArray<WalletTransaction>;
```

Comment:
I suggest to store transactions in a normalized form (by id)

https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape